BetSystem/Endpoint/SportEventEndpoint.cs

```diff
@@ -2,28 +2,105 @@
 using Microsoft.AspNetCore.Mvc;
-using System.Runtime.CompilerServices;
 
 namespace BetSystem.Endpoint
 {
     public static class SportEventEndpoint
     {
-        public static void AddSportEventMapping(this WebApplication app) 
+        public static void AddSportEventMapping(this WebApplication app)
         {
-            app.MapPost("/SportEvent",PostSportEvent);
-            app.MapGet("/SportEvent",GetAllEvents);
-            app.MapGet("/SportEvent/{id}",GetById);
+            app.MapPost("/SportEvent", PostSportEvent);
+            app.MapGet("/SportEvent", GetAllEvents);
+            app.MapGet("/SportEvent/{id}", GetById);
+            app.MapPut("/SportEvent/{id}", PutSportEvent);
             app.MapDelete("/SportEvent/{id}", DeleteById);
+
+            app.MapPost("/SportEvent/{id}/teams", PostTeamToSportEvent);
+            app.MapGet("/SportEvent/{id}/teams", GetTeamsFromSportEvent);
+
+            app.MapPost("/SportEvent/{id}/bets", PlaceBetOnEvent);
+            app.MapGet("/SportEvent/{id}/bets", GetBetToSportEvent);
+
+            app.MapPost("/SportEvent/{id}/results", PostResultToSportEvent);
+            app.MapGet("/SportEvent/{id}/results", GetResultFromSportEvent);
+
         }
 
-        public static IResult PostSportEvent(IValidator<SportEventDto> validator,SportEventDto sportEventDto, [FromServices] ISportEventService service)
+        
+
+        private static IResult GetResultFromSportEvent(int id, [FromServices] ISportEventService service)
         {
-            var result = validator.Validate(sportEventDto);
+            var sportEvent = service.GetById(id);
+            if (sportEvent == null) return Results.NotFound();
+            return Results.Ok(service.GetResultsFromSportEvent(id));
+        }
+
+        private static IResult PostResultToSportEvent(int id, EventResultDto eventResultDto, IValidator<EventResultDto> validator, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+
+            if (sportEvent == null) return Results.NotFound();
+            var result = validator.Validate(eventResultDto);
+
+            if (!result.IsValid) return Results.ValidationProblem(result.ToDictionary());
+            if (service.GetById(id) == null) Results.NotFound(id);
+            if (service.GetById(eventResultDto.TeamId) == null) return Results.BadRequest("Team Not Found");
+
+            service.AddResultToSportEvent(id, eventResultDto);
+
+            return Results.Ok();
+
+        }
+
+        private static IResult GetBetToSportEvent(int id, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+            if (sportEvent == null) return Results.NotFound();
+            return Results.Ok(service.GetBetsFromSportEvent(id));
+        }
+
+        private static IResult PlaceBetOnEvent(int id,IValidator<BetOnEventDto> validator, BetOnEventDto betOnEventDto, [FromServices] ISportEventService service)
```

Comment:
Are service is good name for variable?

Reply:
And in other places it also need to be changed.

Reply:
Changed in all places names for variable